### DISCUSSION TITLE
AuthNavigator not showing after logout in IOS

### DISCUSSION THREAD

**Discussion by bytesagar** [react-navigation/react-navigation#10521] [Q&A]
``` 
"@react-navigation/bottom-tabs": "^6.3.1",
    "@react-navigation/native": "^6.0.8",
    "@react-navigation/native-stack": "^6.5.0",
    "@react-navigation/stack": "^6.1.1",
```

My Navigator
```
const AppRoute = () => {
  const [isReady, setIsReady] = useState(false);
  const [user, setUser] = useState<any>();

  const restoreUser = async () => {
    const user = await getUser();
    user && setUser(user);
  };

  if (!isReady) {
    return (
      <AppLoading
        startAsync={restoreUser}
        onError={(err) => console.log(err)}
        onFinish={() => setIsReady(true)}
      />
    );
  }

  return (
    <AuthContext.Provider value={{ user, setUser }}>
      <NavigationContainer
        theme={{ colors: { background: colors.background } }}
      >
        {user ? <MainNavigator /> : <AuthNavigator />}
      </NavigationContainer>
    </AuthContext.Provider>
  );
};
```

Logout function
```
const logOut = async () => {
    try {
      await removeToken("token");
      await removeItem("@profile");
      await removeItem("@name");
      setUser(null);

      // Toast.show({
      //   text1: "Successfully logged out!",
      //   position: "bottom",
      //   type: "success",
      //   visibilityTime: 1000,
      // });
    } catch (error) {
      console.log(error);
    }
  };
```
When I press logout in IOS it freezes showing black screen. But when I restart it already had logged out user.

**Comment by raajnadar** [comment]
Edit - The issue is you don't have any mounted navigation when you switch the states, your condition to switch auth state should be inside the navigator. Give the below snippet a try and let me know if that solves the issue.

Also, you can take reference from this StackOverflow answer, https://stackoverflow.com/a/68118450/5519872

Example - 

```
<NavigationContainer>
    <Stack.Navigator>
       {!token ? (
          <Stack.Screen name="Login" component={Login} />
       ) : (
         <Stack.Screen name="Home" component={Home} />
       )}
    </Stack.Navigator>
</NavigationContainer>
```


This should not be an issue but still

`const [user, setUser] = useState<any>(null) <--- make this null initially`

Can you add a snack repro to check what is going wrong??

**Reply by bytesagar** [reply]
@raajnadar  I have one AuthStackNavigator and one MainNavigator. while validating then with token how do I get slidefromRight animation on authstack first screen after logout

**Reply by raajnadar** [reply]
By default the navigator uses the os specific animation, to customize use this option https://reactnavigation.org/docs/stack-navigator/#pre-made-configs
